vendor/ocaml/bytecomp/typeopt.ml

```diff
@@ -25,6 +28,48 @@ let has_base_type exp base_ty_path =
   | Tconstr(p, _, _) -> Path.same p base_ty_path
   | _ -> false
 
+
+(**  [Types.constructor_description]
+     records the type at the definition type so for ['a option]
```

Comment:
Nit: I don't understand this comment. Though I understand what the code does.

Reply:
The point is that `Types.constructor_description` does not have enough information, we need typedtree and env